src/configure.rs

```diff
@@ -118,6 +121,40 @@ pub enum ConfigureError {
 
     #[error("That project key is not defined in keys.json")]
     MissingProjectKey,
+
+    #[error("No environment variable or secrets repository found – cannot decrypt files")]
+    MissingDecryptionKey,
+
+    #[error("This decryption key is not valid base64")]
+    DecryptionKeyEncodingError,
+
+    #[error("This decryption key is not a sodium-compatible key")]
+    DecryptionKeyParsingError,
+
+    #[error("Unable to read input file")]
+    InputFileNotReadable,
+
+    #[error("Unable to write output file")]
+    OutputFileNotWritable,
+}
+
+// These implementations should be removed as soon as we can adopt `ConfigureError` everywhere
+impl From<std::io::Error> for ConfigureError {
+    fn from(_error: std::io::Error) -> Self {
+        ConfigureError::DataDecryptionError
+    }
+}
+
+impl From<git2::Error> for ConfigureError {
+    fn from(_error: git2::Error) -> Self {
+        ConfigureError::SecretsRepoError
+    }
+}
+
+impl From<std::num::ParseIntError> for ConfigureError {
+    fn from(_error: std::num::ParseIntError) -> Self {
+        ConfigureError::GitStatusParsingError
+    }
```

Comment:
> Don't we run the risk of converting these more generic error types to an invalid ConfigureError type

This certainly _could_ be an issue, though I felt the risk wasn't terribly high.

IMHO the most problematic one is `std::io::Error` – that could be a _lot_ of things.

The `git2:Error` issues can only arise when working with the secrets repo. Though given the fact that we can't use `git2` everywhere, I'm inclined to tear it out and replace it with command-line invocations.

The last error is used only one time for some very specific git work, so that one is not very concerning to me

Reply:
I guess for now, we can consider it just as something to keep in mind for future. 🤷